drivers/include/ph_oem.h

```diff
@@ -0,0 +1,356 @@
+    ph_oem_irq_option_t irq_option; /**< behavior of the interrupt pin, disabled by default*/
+} ph_oem_params_t;
+
+/**
+ * @brief   pH OEM interrupt pin callback
+ */
+typedef void (*ph_oem_interrupt_pin_cb_t)(void *);
+
+/**
+ * @brief   pH OEM device descriptor
+ */
+typedef struct ph_oem {
+    ph_oem_params_t params;         /**< device driver configuration */
+    ph_oem_interrupt_pin_cb_t cb;   /**< interrupt pin callback */
+    void *arg;                      /**< interrupt pin callback param */
+} ph_oem_t;
+
+/**
+ * @brief   Initialize a pH OEM sensor
+ *
+ * @param[in,out]   dev      device descriptor
+ * @param[in]       params   device configuration
+ *
+ * @return @ref PH_OEM_OK on success
+ * @return @ref PH_OEM_NODEV if no device is found on the bus
+ * @return @ref PH_OEM_NOT_PH if the device found at the address is not a pH OEM device
+ * @return
+ */
+int ph_oem_init(ph_oem_t *dev, const ph_oem_params_t *params);
+
+/**
+ * @brief   Sets a new address to the pH OEM device by unlocking the
+ *          @ref PH_OEM_REG_UNLOCK register and  writing a new address to
+ *          the @ref PH_OEM_REG_ADDRESS register.
+ *          The device address will also be updated in the device descriptor so
+ *          it is still usable.
+ *
+ *          Settings are retained in the sensor if the power is cut.
+ *
+ *          The address in the device descriptor will reverse to the default
+ *          address you provided through PH_OEM_PARAM_ADDR after the
+ *          microcontroller restarts
+ *
+ * @param[in] dev   device descriptor
+ * @param[in] addr  new address for the device. Range: 0x01 - 0x7f
+ *
+ * @return @ref PH_OEM_OK on success
+ * @return @ref PH_OEM_WRITE_ERR if writing to the device failed
+ */
+int ph_oem_set_i2c_address(ph_oem_t *dev, uint8_t addr);
+
+/**
+ * @brief   Enable the pH OEM interrupt pin if @ref ph_oem_params_t.interrupt_pin
+ *          is defined.
+ *          @note Provide the PH_OEM_PARAM_INTERRUPT_OPTION flag in your
+ *          makefile. Valid options see: @ref ph_oem_irq_option_t.
+ *          The default is @ref PH_OEM_IRQ_BOTH
+ *
+ * @param[in] dev       device descriptor
+ * @param[in] cb        callback called when the pH OEM interrupt pin fires
+ * @param[in] arg       callback argument
+ * @param[in] gpio_mode @ref gpio_mode_t <br>
+ *                      most likely @ref GPIO_IN. If the pin is to sensitive use
+ *                      @ref GPIO_IN_PU for @ref PH_OEM_IRQ_FALLING or <br>
+ *                      @ref GPIO_IN_PD for @ref PH_OEM_IRQ_RISING and
+ *                      @ref PH_OEM_IRQ_BOTH
+ *
+ * @return @ref PH_OEM_OK on success
+ * @return @ref PH_OEM_WRITE_ERR if writing to the device failed
+ * @return @ref PH_OEM_INTERRUPT_GPIO_UNDEF if the interrupt pin is undefined
+ * @return @ref PH_OEM_GPIO_INIT_ERR if initializing the interrupt gpio pin failed
+ */
+int ph_oem_enable_interrupt(ph_oem_t *dev, ph_oem_interrupt_pin_cb_t cb,
```

Comment:
Now that the interrupt option has been moved to the params and as the gpio mode is not likely to change during runtime maybe it can also be moved there (as it depends on the selected interrupt option).

Reply:
I have moved the gpio mode to the params